tracer/src/Datadog.Trace/Debugger/ExceptionAutoInstrumentation/ExceptionTrackManager.cs

```diff
@@ -0,0 +1,513 @@
+
+namespace Datadog.Trace.Debugger.ExceptionAutoInstrumentation
+{
+    internal class ExceptionTrackManager
+    {
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor<ExceptionTrackManager>();
+        private static readonly ConcurrentDictionary<ExceptionIdentifier, TrackedExceptionCase> TrackedExceptionCases = new();
+        private static readonly ConcurrentQueue<Exception> ExceptionProcessQueue = new();
+        private static readonly SemaphoreSlim WorkAvailable = new(0, int.MaxValue);
+        private static readonly CancellationTokenSource Cts = new();
+        private static readonly ExceptionCaseScheduler ExceptionsScheduler = new();
+        private static readonly int MaxFramesToCapture = ExceptionDebugging.Settings.MaximumFramesToCapture;
+        private static readonly TimeSpan RateLimit = ExceptionDebugging.Settings.RateLimit;
+        private static readonly BasicCircuitBreaker ReportingCircuitBreaker = new(ExceptionDebugging.Settings.MaxExceptionAnalysisLimit, TimeSpan.FromSeconds(1));
+        private static Task _exceptionProcessorTask;
+        private static bool _isInitialized;
+
+        private static async Task StartExceptionProcessingAsync(CancellationToken cancellationToken)
+        {
+            while (!cancellationToken.IsCancellationRequested)
+            {
+                await WorkAvailable.WaitAsync(cancellationToken).ConfigureAwait(false);
+
+                while (ExceptionProcessQueue.TryDequeue(out var exception))
+                {
+                    ProcessException(exception, ErrorOriginKind.HttpRequestFailure, rootSpan: null);
+                }
+            }
+        }
+
+        public static void Report(Span span, Exception exception)
+        {
+            if (!_isInitialized)
+            {
+                Log.Information(exception, "Exception Track Manager is not initialized yet. Skipping the processing of an exception. Span = {Span}", span);
+                return;
+            }
+
+            // For V1 of Exception Debugging, we only care about exceptions propagating up the stack
+            // and marked as error by the service entry/root span.
+            if (!span.IsRootSpan || exception == null || !IsSupportedExceptionType(exception))
+            {
+                Log.Information(exception, "Skipping the processing of the exception. Span = {Span}", span);
+                return;
+            }
+
+            try
+            {
+                ReportInternal(exception, ErrorOriginKind.HttpRequestFailure, span);
+            }
+            catch (Exception ex)
+            {
+                Log.Error(ex, "An exception was thrown while processing an exception for tracking.");
+            }
+        }
+
+        private static void ReportInternal(Exception exception, ErrorOriginKind errorOrigin, Span rootSpan)
+        {
+            if (CachedDoneExceptions.Contains(exception.ToString()))
```

Comment:
why not? 🤔 
Also, the code of the tracer that calls into _exception debugging_ calls ToString on the exception object to dump the stacktrace as a tag on the service root span.

Reply:
> why not?

Just FYI 😅 
- https://github.com/DataDog/dd-trace-dotnet/pull/5291

Reply:
Nice! Well, since I'm entering the mentioned call path _right after_ the exception tags are added (alongside `exception.ToString()`, then I can simply assume `ToString` is fine. In other words, if there is no `error.stack` tag - Exception Debugging should not act on the span (as the whole mechanism is based on Error Tracking that drinks all those spans that are marked with errors).